great_expectations/rule_based_profiler/domain_builder/column_domain_builder.py

```diff
@@ -179,26 +179,16 @@ def get_effective_column_names(
         validator: Optional["Validator"] = None,  # noqa: F821
         variables: Optional[ParameterContainer] = None,
     ) -> List[str]:
-        # Obtain include_column_names from "rule state" (i.e., variables and parameters); from instance variable otherwise.
-        include_column_names: Optional[
-            List[str]
-        ] = get_parameter_value_and_validate_return_type(
-            domain=None,
-            parameter_reference=self.include_column_names,
-            expected_return_type=None,
-            variables=variables,
-            parameters=None,
-        )
-
-        # Obtain exclude_column_names from "rule state" (i.e., variables and parameters); from instance variable otherwise.
-        exclude_column_names: Optional[
-            List[str]
-        ] = get_parameter_value_and_validate_return_type(
-            domain=None,
-            parameter_reference=self.exclude_column_names,
-            expected_return_type=None,
-            variables=variables,
-            parameters=None,
+        """
+        This method applies multiple directives to obtain columns to be included as part of returned "Domain" objects.
+        """
+        include_column_names: Optional[List[str]] = cast(
```

Comment:
The return type of the _resolve_list_type_property is an empty list (and never None), so we can remove the Optional type hint here.

Reply:
@donaldheppner Yes, we can.  In next commit.

Reply:
@donaldheppner This was done (committed).